pyproject.toml

```diff
@@ -93,11 +94,11 @@ multidict = ">=6.0.2"
 opentelemetry-instrumentation-asgi = { version = "*", optional = true }
 picologging = { version = "*", optional = true }
 polyfactory = ">=2.3.2"
-prometheus-client = {version = "*", optional = true}
-pydantic = "<2"
-python-dateutil = { version = "*", optional = true }
+prometheus-client = { version = "*", optional = true }
+pydantic = "*"
+python-dateutil = "*"
```

Comment:
Why is this required now?

Reply:
we relied on the pydantic date-time parsers for parsing of path params (in v1 they vendored these from django). thats removed in pydantic v2, and anyhow we dont want to rely on them. We either do the above or we introduce a breaking change for path params, which we might do - but not in this scope.